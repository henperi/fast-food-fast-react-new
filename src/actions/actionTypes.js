export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';
export const SIGNUP_USER = 'SIGNUP_USER';
export const SET_AUTH_USER = 'SET_AUTH_USER';
export const REMOVE_AUTH_USER = 'REMOVE_AUTH_USER';
export const SET_ERRORS = 'SET_ERRORS';
export const GET_MENU = 'GET_MENU';
export const ADD_TO_CART = 'ADD_TO_CART';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const UPDATE_CART_ITEM_QUANTITY = 'UPDATE_CART_ITEM_QUANTITY';
export const TOGGLE_CART_SLIDER = 'TOGGLE_CART_SLIDER';
export const FETCH_CART_ITEMS = 'FETCH_CART_ITEMS';
export const SET_VISIBLE_CART = 'SET_VISIBLE_CART';
