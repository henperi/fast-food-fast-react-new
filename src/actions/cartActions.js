import {
  ADD_TO_CART, REMOVE_FROM_CART, TOGGLE_CART_SLIDER, SET_CURRENT_CART,
} from './actionTypes';

export const setCurrentCart = payload => ({
  type: SET_CURRENT_CART,
  payload,
});

export const addToCart = (id, name, price) => ({
  type: ADD_TO_CART,
  id,
  name,
  price,
});

export const removeFromCart = id => ({ type: REMOVE_FROM_CART, id });

export const updateCartQty = (id, updates) => ({
  type: 'UPDATE_CART_ITEM_QUANTITY',
  updates,
  id,
});

export const emptyCart = () => {
  localStorage.removeItem('foodItems');
  return { type: 'EMPTY_CART' };
};

export const toggleCartSlider = () => ({ type: TOGGLE_CART_SLIDER });
