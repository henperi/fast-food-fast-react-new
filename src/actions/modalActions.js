// import { OPEN_MODAL } from './actionTypes';

// export const openModal = ({
//   id, cat, title, price, description,
// } = {}) => ({
//   type: OPEN_MODAL,
//   modal: {
//     isOpen: true,
//     id,
//     cat,
//     title,
//     price,
//     description,
//   },
// });

// export const emptyCart = () => ({ type: 'EMPTY_CART' });
