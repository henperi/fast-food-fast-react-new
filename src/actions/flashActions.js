import { SHOW_FLASH, REMOVE_FLASH } from './actionTypes';

export const showFlash = (data = {}) => {
  const { title = '', message = '', flashType = 'success' } = data;
  return {
    type: SHOW_FLASH,
    payload: {
      title,
      message,
      flashType,
    },
  };
};

export const removeFlash = () => ({
  type: REMOVE_FLASH,
});
