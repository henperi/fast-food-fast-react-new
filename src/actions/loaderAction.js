import { TOGGLE_LOADER } from './actionTypes';

const toggleLoader = () => ({
  type: TOGGLE_LOADER,
});

export default toggleLoader;
