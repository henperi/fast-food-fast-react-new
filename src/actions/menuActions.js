import { SET_MENU, SET_ERRORS, SET_FEATURED_MENU } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';

export const setMenu = (payload = {}) => ({
  type: SET_MENU,
  payload,
});

export const setFeaturedMenu = (payload = {}) => {
  const featuredMenu = payload.slice(0, 3);

  return {
    type: SET_FEATURED_MENU,
    payload: featuredMenu,
  };
};

export const getMenu = () => (dispatch) => {
  axiosInstance
    .get('/menu')
    .then((response) => {
      const { data } = response;
      dispatch({
        type: SET_MENU,
        payload: { menu: [...data.menu] },
      });
      dispatch(setFeaturedMenu(data.menu));
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export const updateMenu = () => {};
