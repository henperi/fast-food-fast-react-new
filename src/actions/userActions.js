import axiosInstance, { setAxiosToken } from '../utils/axiosInstance';
import { SET_ERRORS, SET_AUTH_USER, REMOVE_AUTH_USER } from './actionTypes';

export const setAuthUser = (payload = {}) => ({
  type: SET_AUTH_USER,
  payload,
});

export const logoutUser = () => ({
  type: REMOVE_AUTH_USER,
  payload: {
    isAuthenticated: false,
  },
});

const signupUrl = '/auth/signup';
const loginUrl = '/auth/login';

export const signupUser = (userData = {}, history) => (dispatch) => {
  axiosInstance
    .post(signupUrl, userData)
    .then((response) => {
      const { data } = response;
      const isAuthenticated = true;
      const { userToken } = data;
      const { userId, role } = data.createdUser;
      const name = data.createdUser.fullname;

      localStorage.setItem('userToken', userToken);
      localStorage.setItem('userId', userId);
      localStorage.setItem('fullname', name);
      localStorage.setItem('role', role);

      setAxiosToken(userToken);

      dispatch({
        type: SET_AUTH_USER,
        payload: {
          isAuthenticated,
          userId,
          userToken,
          role,
          fullname: name,
        },
      });

      return history.push('/users/foods');
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      // console.log(response);
      // console.log('=======');
      // console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export const loginUser = (userData = {}, history) => (dispatch) => {
  axiosInstance
    .post(loginUrl, userData)
    .then((response) => {
      // console.log('=============', response, '=============');
      const { data } = response;
      const isAuthenticated = true;
      const { userToken } = data;
      const { fullname, role, userId } = data.userData;

      localStorage.setItem('userToken', userToken);
      localStorage.setItem('fullname', fullname);
      localStorage.setItem('role', role);
      localStorage.setItem('userId', userId);

      setAxiosToken(userToken);

      dispatch(
        setAuthUser({
          isAuthenticated,
          fullname,
          role,
          userId,
          userToken,
        }),
      );

      return history.push('/users/foods');
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      // console.log(response);
      // console.log('=======');
      // console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};
