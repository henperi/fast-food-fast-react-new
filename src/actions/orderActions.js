import { SET_ERRORS } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';
import { emptyCart } from './cartActions';

const placeOrder = (cartItems = {}, handleModal) => (dispatch) => {
  console.log(cartItems);
  const orderData = {
    foodItems: cartItems,
  };
  axiosInstance
    .post('/orders', orderData)
    .then((response) => {
      const { data } = response;
      console.log(data);
      handleModal();
      dispatch(emptyCart());
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(errors);
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export default placeOrder;
