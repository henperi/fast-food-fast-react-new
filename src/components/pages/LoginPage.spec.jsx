/* eslint-disable no-undef */
import React from 'react';
import sinon from 'sinon';
import { LoginPage } from './LoginPage';
import { loginUser } from '../../actions/userActions';
// import SignupForm from './SignupForm';

describe('The Signup Component Test Suite', () => {
  const user = {};
  const wrapper = shallow(<LoginPage loginUser={loginUser} user={user} />);
  it('should render the <SignupForm /> Component', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(LoginPage)).toBeDefined();
  });

  it('should handle input change event for email', () => {
    const event = {
      target: { name: 'email', value: 'henry.izontimi@gmail.com' },
      preventDefault: () => {},
    };

    wrapper.instance().onChange(event);
    expect(wrapper.state().email).toEqual('henry.izontimi@gmail.com');
    expect(wrapper.instance().state.email).toEqual(event.target.value);
  });

  it('should handle input change event for password', () => {
    const event = {
      target: { name: 'password', value: '12345qwe' },
      preventDefault: () => {},
    };

    wrapper.instance().onChange(event);
    expect(wrapper.state().password).toEqual('12345qwe');
    expect(wrapper.instance().state.password).toEqual(event.target.value);
  });

  it('should call the onSubmit', () => {
    const event = { preventDefault: () => {} };

    wrapper.instance().onSubmit(event);
    expect(wrapper.state().email).toEqual('henry.izontimi@gmail.com');
    expect(wrapper.state().password).toEqual('12345qwe');
    expect(sinon.stub(jest.fn(wrapper.instance().props.loginUser()))).not.toHaveBeenCalled();
    // expect(
    //   sinon.stub(jest.fn(wrapper.instance().props.actions(reportArticle()))),
    // ).toHaveBeenCalled();
    // expect(wrapper.instance().state.reportType).toEqual(event.target.value);
  });

  it('should render exactly one form tag', () => {
    expect(wrapper.find('form')).toHaveLength(1);
  });
  it('should render exactly two input tags', () => {
    expect(wrapper.find('input')).toHaveLength(2);
  });
});
