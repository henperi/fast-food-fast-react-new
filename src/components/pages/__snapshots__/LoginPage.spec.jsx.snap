// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Signup Component Test Suite should render the <SignupForm /> Component 1`] = `
<main>
  <div
    className="container hero-fullx"
  >
    <div
      className="text-center contain-50"
    >
      <div
        className="card card-shadow text-inverse"
      >
        <h2>
          Login To Your Account
        </h2>
        <p>
          Login to have a wonderful experience and enjoy amazing discounts
        </p>
        <form
          className="form-data contain"
          onSubmit={[Function]}
        >
          <div
            className="form-member"
          >
            <input
              className="form-input email"
              name="email"
              onChange={[Function]}
              placeholder="Enter Email"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-member"
          >
            <input
              className="form-input password"
              name="password"
              onChange={[Function]}
              placeholder="Enter Password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-member"
          >
            <div
              className="response-area"
            />
            <div
              className="loader hide"
            >
              <i
                className="fa fa-spinner fa-spin"
              />
            </div>
            <button
              className="btn btn-green btn-block login"
              name="login"
              type="submit"
            >
              Login
            </button>
          </div>
          <p>
            Dont have an account?
            <Link
              className="link"
              replace={false}
              to="/signup"
            >
              Signup Instead
            </Link>
          </p>
          <p>
            Go back home
            <Link
              className="link"
              replace={false}
              to="/"
            >
              Home
            </Link>
          </p>
        </form>
      </div>
    </div>
  </div>
</main>
`;
